src/registry-auth-locator/config.test.ts

```diff
@@ -0,0 +1,35 @@
+import { DockerConfig } from "./types";
+
+import fs from "fs";
+import { readDockerConfig } from "./config";
+
+const spy = jest.spyOn(fs, "existsSync").mockImplementation();
+
+describe("Config", () => {
+  afterEach(() => (process.env.DOCKER_AUTH_CONFIG = undefined));
+
+  it("should return empty object if no config file exists", async () => {
+    spy.mockReturnValue(false);
+
+    const config = await readDockerConfig();
+    expect(config).toEqual({});
+  });
+
+  it("should use the docker environment variable config", async () => {
+    const dockerConfig: DockerConfig = {
+      auths: {
+        "https://registry.example.com": {
+          email: "user@example.com",
+          username: "user",
+          password: "pass",
+        },
+      },
+    };
+    process.env.DOCKER_AUTH_CONFIG = JSON.stringify(dockerConfig);
+
+    spy.mockReturnValue(false);
+
+    const config = await readDockerConfig();
+    expect(config).toEqual(dockerConfig);
+  });
```

Comment:
I don't suppose you could add the remaining case, where the file does exist? I guess you could spy on `readFile`? I know it's not introduced by your change but it'd complete the test spec 🙂

Reply:
Ofcourse, I didn't want to do it before I knew your thoughts on these changes in general. :)